guava/src/com/google/common/eventbus/SubscriberRegistry.java

```diff
@@ -183,11 +190,20 @@ void unregister(Object listener) {
           Class<?>[] parameterTypes = method.getParameterTypes();
           checkArgument(
               parameterTypes.length == 1,
-              "Method %s has @Subscribe annotation but has %s parameters."
+              "Method %s has @Subscribe annotation but has %s parameters. "
                   + "Subscriber methods must have exactly 1 parameter.",
               method,
               parameterTypes.length);
 
+          checkArgument(
+              !parameterTypes[0].isPrimitive(),
+              "@Subscribe method %s's parameter is %s. "
+                  + "Subscriber methods cannot accept primitives. "
+                  + "Consider changing the parameter to %s.",
+              method,
+              parameterTypes[0].getName(),
+              Primitives.wrap(parameterTypes[0]).getSimpleName());
```

Comment:
Hi :)

Can we somehow postpone the string concatenation and param wrapping for only when the argument is illegal?

For example, what log4j do.
Or by using a lambda (I know lambdas aren't welcome on Guava. See one more option at the bottom), like:

```
Object o = lazyToString(() -> someLongCalculatedString());

public static Object lazyToString(Supplier<String> supplier)
{
	return new Object()
	{
		@Override
		public String toString()
		{
			return supplier.get();
		}
	};
}

```

Another option is to just check the argument inline and call a specific method to throw the `IllegalArgumentException`.
Example:
```
if (parameterTypes[0].isPrimitive()) {
  throwIllegalArgumentException(
          "@Subscribe method %s's parameter is %s. "
              + "Subscriber methods cannot accept primitives. "
              + "Consider changing the parameter to %s.",
          method,
          parameterTypes[0].getName(),
          Primitives.wrap(parameterTypes[0]).getSimpleName());
}
```

What do you think?
@netdpb 

Cheers.

Reply:
Come to think about it, 3rd option looks even more readable than `checkArgument()`.